tests/test_requests.py

```diff
@@ -1203,6 +1203,46 @@ def test_request_and_response_are_pickleable(self, httpbin):
         assert r.request.url == pr.request.url
         assert r.request.headers == pr.request.headers
 
+    def test_response_lines(self):
+        """
+        iter_lines should be able to handle data dribbling in which might
+        not be lined up ideally.
+        """
+        mock_chunks = [
+            'This \r\n',
+            '',
+            'is\r',
+            '\n',
+            'a',
+            ' ',
+            '',
+            '',
+            'test.',
+            '\r',
+            '\n',
+            'end.',
+        ]
+
+        def mock_iter_content(*args, **kwargs):
+            '''Fake difficult data.'''
+            for chunk in mock_chunks:
+                yield chunk
+
+        r = requests.Response()
+        r._content_consumed = True
+        r.iter_content = mock_iter_content
+
+        assert list(r.iter_lines(delimiter='\r\n')) == \
+            ''.join(mock_chunks).split('\r\n')
```

Comment:
I'd like these asserts cleaned up if at all possible, they're kinda complex. Let's break some stuff up so the assert is somewhat self-describing.